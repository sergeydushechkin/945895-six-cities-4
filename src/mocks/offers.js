const TYPES = {
  apartment: `Apartment`,
  room: `Private room`,
  house: `House`,
  hotel: `Hotel`
};

export default [
  {
    id: 1,
    city: `Amsterdam`,
    title: `Beautiful & luxurious apartment at great location`,
    pictures: [`img/apartment-01.jpg`, `img/room.jpg`, `img/apartment-02.jpg`, `img/apartment-03.jpg`, `img/studio-01.jpg`, `img/apartment-01.jpg`],
    price: 120,
    type: TYPES.apartment,
    rating: 4.8,
    isPremium: true,
    isFavorite: false,
    bedrooms: 2,
    descriptions: [
      `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`,
      `An independent House, strategically located between Rembrand Square and National Opera, but where the bustle of the city comes to rest in this alley flowery and colorful.`
    ],
    guests: 3,
    features: [`Wi-Fi`, `Washing machine`, `Towels`, `Heating`, `Coffee machine`, `Baby seat`, `Kitchen`, `Dishwasher`, `Cabel TV`, `Fridge`],
    hostId: 1,
    coordinates: [52.3909553943508, 4.85309666406198]
  },
  {
    id: 2,
    city: `Amsterdam`,
    title: `Wood and stone place`,
    pictures: [`img/room.jpg`, `img/apartment-02.jpg`, `img/apartment-03.jpg`, `img/studio-01.jpg`, `img/apartment-01.jpg`],
    price: 80,
    type: TYPES.room,
    rating: 4.2,
    isPremium: false,
    isFavorite: true,
    bedrooms: 1,
    descriptions: [`Large wood and stone place in Amsterdam`],
    guests: 2,
    features: [`Wi-Fi`, `Washing machine`, `Towels`, `Heating`],
    hostId: 3,
    coordinates: [52.369553943508, 4.85309666406198]
  },
  {
    id: 3,
    city: `Amsterdam`,
    title: `Canal View Prinsengracht`,
    pictures: [`img/apartment-02.jpg`, `img/apartment-03.jpg`, `img/studio-01.jpg`, `img/apartment-01.jpg`],
    price: 132,
    type: TYPES.apartment,
    rating: 4,
    isPremium: false,
    isFavorite: false,
    bedrooms: 2,
    descriptions: [`A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.`],
    guests: 1,
    features: [`Washing machine`, `Towels`, `Heating`, `Coffee machine`, `Baby seat`, `Kitchen`],
    hostId: 4,
    coordinates: [52.3909553943508, 4.929309666406198]
  },
  {
    id: 4,
    city: `Amsterdam`,
    title: `Nice, cozy, warm big bed apartment`,
    pictures: [`img/apartment-03.jpg`, `img/apartment-02.jpg`, `img/apartment-01.jpg`, `img/room.jpg`, `img/studio-01.jpg`, `img/apartment-03.jpg`],
    price: 180,
    type: TYPES.apartment,
    rating: 5,
    isPremium: true,
    isFavorite: false,
    bedrooms: 4,
    descriptions: [
      `The building is green and from 18th century.`, `An independent House, strategically located between Rembrand Square and National Opera.`
    ],
    guests: 3,
    features: [`Wi-Fi`, `Washing machine`, `Baby seat`, `Kitchen`, `Dishwasher`, `Cabel TV`, `Fridge`],
    hostId: 1,
    coordinates: [52.3809553943508, 4.939309666406198]
  },
];
