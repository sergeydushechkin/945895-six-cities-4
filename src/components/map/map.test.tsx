import * as React from "react";
import * as renderer from "react-test-renderer";
import Main from "./map";

import offers from "../../mocks/tests-offers";

it(`Render Map`, () => {
  const tree = renderer
    .create(
        <Main
          city={[52.38333, 4.9]}
          offers={offers}
          activeOfferId={1}
          className={`cities__map map`}
          zoom={10}
        />,
        {
          createNodeMock: () => {
            return document.createElement(`div`);
          }
        }
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
