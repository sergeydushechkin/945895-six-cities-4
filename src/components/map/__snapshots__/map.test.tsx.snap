// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Map 1`] = `
<section
  className="cities__map map"
/>
`;
